features/lwipstack/LWIPStack.cpp

```diff
@@ -488,12 +488,18 @@ nsapi_size_or_error_t LWIP::socket_sendto(nsapi_socket_t handle, const SocketAdd
 {
     struct mbed_lwip_socket *s = (struct mbed_lwip_socket *)handle;
     ip_addr_t ip_addr;
+    nsapi_addr_t interface_addr;
 
     nsapi_addr_t addr = address.get_addr();
     if (!convert_mbed_addr_to_lwip(&ip_addr, &addr)) {
         return NSAPI_ERROR_PARAMETER;
     }
 
+    convert_lwip_addr_to_mbed(&interface_addr, get_ip_addr(true, &default_interface->netif));
+    if(addr.version != interface_addr.version){
```

Comment:
Does this work in dual stack case when you have both addresses? `get_ip_addr` will return the preferred address (defined with `lwip.ip-ver-pref` configuration), but the application uses the other address version?

Reply:
If there is both IP4 and IP6 IP_VERSION_PREF   must be properly  set.

Reply:
In order to get rid of prefered dependency, how about something like this:

lwip_tools.cpp:
```
bool LWIP::dualstack_enabled()
{
#if LWIP_IPV4 && LWIP_IPV6
    if (get_ipv4_addr(netif) && get_ipv6_addr(netif)) {
        return true;
    }
#endif
    return false;
}
```
and then in version check first check that dualstack is on:
`if (dualstack_enabled() && addr.version != interface_addr.version) {`

Reply:
Modified